AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -198,24 +198,15 @@ private static void saveRequestForLater(Map<String, Object> args) {
       SharedPreferences preferences = context.getSharedPreferences(
           LEANPLUM, Context.MODE_PRIVATE);
       SharedPreferences.Editor editor = preferences.edit();
-      int count = preferences.getInt(Constants.Defaults.COUNT_KEY, 0);
-      String itemKey = String.format(Locale.US, Constants.Defaults.ITEM_KEY, count);
-      if (count == MAX_KEY_COUNT - 1) {
-        count = 0;
-      } else {
-        count++;
-      }
-
+      long count = LeanplumEventDataManager.getEventsCount();
       String uuid = preferences.getString(Constants.Defaults.UUID_KEY, null);
-      if (uuid == null) {
+      if (uuid == null || count % MAX_ACTIONS_PER_API_CALL == 0) {
```

Comment:
the new uuid will be generated when we reach 10k events ? what happens if there is never 10k events exactly ?